src/app/shared/services/student.service.ts

```diff
@@ -0,0 +1,145 @@
+import { Injectable, Signal, WritableSignal, computed, inject, signal } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+
+import {  Observable, Subject, defer } from 'rxjs';
+import { tap, exhaustMap } from 'rxjs/operators';
+
+import { Firestore, collection, orderBy, query, addDoc, updateDoc, serverTimestamp, deleteDoc, doc, DocumentReference, DocumentData } from 'firebase/firestore';
+import { collectionData } from 'rxfire/firestore';
+
+
+import { FIRESTORE } from '../../app.config';
+import { Student } from '../interfaces/students';
+
+interface StudentState {
+  students: Student[];
+  error: string | null;
+  loading: boolean;
+  selectedStudent: Student | null;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class StudentService {
+  private fireStore: Firestore = inject(FIRESTORE);
+
+  private students$: Observable<Student[]> = this.getStudents();
+  public addStudent$ = new Subject<Partial<Student>>();
+  public deleteStudent$ = new Subject<string>();
+  public updateStudent$ = new Subject<Partial<Student>>();
+  public selectStudent$ = new Subject<Student>();
+
+  private state: WritableSignal<StudentState> = signal<StudentState>({
+    students: [],
+    error: null,
+    loading: false,
+    selectedStudent: null
+  });
+
+  public students: Signal<Student[]> = computed(() => this.state().students);
+  public loading: Signal<boolean> = computed(() => this.state().loading);
+  public error: Signal<string | null> = computed(() => this.state().error);
+  public selectedStudent: Signal<Student | null> = computed(() => this.state().selectedStudent);
+
+
+  constructor() {
```

Comment:
Let's avoid using constructor of services.  Here you use it for initializing subscriptions, but it is better to be done in ngOnInit of component that uses it.
Create method like initSubscribtions() were you subscribe on your subjects and call it in StudentsComponent